include/unifex/get_stop_token.hpp

```diff
@@ -16,15 +16,44 @@
 #pragma once
 
 #include <unifex/stop_token_concepts.hpp>
+#include <unifex/inplace_stop_token.hpp>
 #include <unifex/tag_invoke.hpp>
 #include <unifex/type_traits.hpp>
+#include <unifex/coroutine.hpp>
 #include <unifex/unstoppable_token.hpp>
+#include <unifex/await_transform.hpp>
 
 #include <unifex/detail/prologue.hpp>
 
 namespace unifex {
 namespace _get_stop_token {
   inline const struct _fn {
+  private:
+#if !UNIFEX_NO_COROUTINES
+    template <typename StopToken>
+    struct awaiter_ {
+      StopToken stoken_;
+      bool await_ready() const noexcept {
+        return false;
+      }
+      template <typename Promise>
+      bool await_suspend(coro::coroutine_handle<Promise> coro) noexcept {
+        stoken_ = _fn{}(coro.promise());
```

Comment:
You don't need this line any more. Retrieval of the stop-token is done in `await_transform()` customisation below.